Specs/Scene/ImplicitSubtreeSpec.js

```diff
@@ -0,0 +1,1599 @@
+      for (var j = 0; j < availability.lengthBits; j++) {
+        expect(subtree.contentIsAvailable(j, i)).toEqual(
+          expectedAvailability[j]
+        );
+      }
+    }
+  }
+
+  function expectChildSubtreeAvailability(subtree, availability) {
+    var expectedAvailability = availabilityToBooleanArray(availability);
+    for (var i = 0; i < availability.lengthBits; i++) {
+      expect(subtree.childSubtreeIsAvailable(i)).toEqual(
+        expectedAvailability[i]
+      );
+    }
+  }
+
+  // used for spying on ResourceCache.load()
+  function fakeLoad(arrayBuffer) {
+    return function (options) {
+      var fakeCacheResource = {
+        typedArray: arrayBuffer,
+      };
+      options.resourceLoader._promise = {
+        promise: when.resolve(fakeCacheResource),
+      };
+    };
+  }
+
+  var tilesetResource = new Resource({
+    url: "https://example.com/tileset.json",
+  });
+  var subtreeResource = new Resource({
+    url: "https://example.com/test.subtree",
+  });
+  var mockTileset = {};
+
+  var quadtreeCoordinates = new ImplicitTileCoordinates({
+    subdivisionScheme: ImplicitSubdivisionScheme.QUADTREE,
+    level: 0,
+    x: 0,
+    y: 0,
+  });
+  var implicitQuadtreeJson = {
+    geometricError: 500,
+    refine: "ADD",
+    boundingVolume: {
+      region: [0, 0, Math.PI / 24, Math.PI / 24, 0, 1000.0],
+    },
+    content: {
+      uri: "https://example.com/{level}/{x}/{y}.b3dm",
+    },
+    extensions: {
+      "3DTILES_implicit_tiling": {
+        subdivisionScheme: "QUADTREE",
+        subtreeLevels: 2,
+        maximumLevel: 1,
+        subtrees: {
+          uri: "https://example.com/{level}/{x}/{y}.subtree",
+        },
+      },
+    },
+  };
+  var implicitQuadtree = new ImplicitTileset(
+    mockTileset,
+    tilesetResource,
+    implicitQuadtreeJson
+  );
+
+  var octreeCoordinates = new ImplicitTileCoordinates({
+    subdivisionScheme: ImplicitSubdivisionScheme.OCTREE,
+    level: 0,
+    z: 0,
+    x: 0,
+    y: 0,
+  });
+  var implicitOctree = new ImplicitTileset(mockTileset, tilesetResource, {
+    geometricError: 500,
+    refine: "REPLACE",
+    boundingVolume: {
+      region: [0, 0, Math.PI / 24, Math.PI / 24, 0, 1000.0],
+    },
+    content: {
+      uri: "https://example.com/{level}/{x}_{y}_{z}.b3dm",
+    },
+    extensions: {
+      "3DTILES_implicit_tiling": {
+        subdivisionScheme: "OCTREE",
+        subtreeLevels: 2,
+        maximumLevel: 3,
+        subtrees: {
+          uri: "https://example.com/{level}/{x}_{y}_{z}.subtree",
+        },
+      },
+    },
+  });
+
+  var internalQuadtreeDescription = {
+    tileAvailability: {
+      descriptor: "11010",
+      lengthBits: 5,
+      isInternal: true,
+    },
+    contentAvailability: [
+      {
+        descriptor: "11000",
+        lengthBits: 5,
+        isInternal: true,
+      },
+    ],
+    childSubtreeAvailability: {
+      descriptor: "1111000010100000",
```

Comment:
the `implicitQuadtreeJson` says `maximumLevel: 1` yet several tests are using `childSubtreeAvailability`. It looks like the implicit tiling code doesn't do anything with `maximumLevel` besides as a getter. Should `getTileIndex` and things like that be aware of this?

Reply:
I'll take a look at these unit tests to see what a better value would be. 

since `getTileIndex` does do bounds checking, it probably is one place where we can put a maximumLevel check. Not sure about other places, validation would ideally be done with a separate tool.

Reply:
Actually, I was talking about this with @lilleyse. The spec doesn't define the behavior when `maximumLevel` is in the middle of the subtree. However, we would want the availability bitstream to still be the full length and just filled with 0s past the maximum level. I made a note of this in an internal issue for tracking spec changes.

The implementation then doesn't need to bounds check against `maximumLevel`, as the availability will just return `false` if you go too deep.

Reply:
also I just pushed a commit to set this maximumLevel artificially high and put a note that this field isn't checked at runtime.